### PR TITLE
[docs] Fix Date Pickers usage to Title Case

### PR DESCRIPTION
**Pull request by richbustos**
### What? 
Updating Date Picker usage to Title Case. 
For example, `Date picker` => `Date Picker`

Work on #9318

### CODE REVIEW

**Review by LukasTy — COMMENTED** [review]
I can spot a few other cases where, **pickers** is spelled with lowercase when searching for _date and time pickers_.
Maybe we should adjust all the instances? 🤔 
Also, the subheading [in here](https://mui.com/x/api/date-pickers/) should probable be **Date and Time Pickers** as well? 🤔

**Comment by richbustos** [comment]
> I can spot a few other cases where, **pickers** is spelled with lowercase when searching for _date and time pickers_.
> Maybe we should adjust all the instances? 🤔
> Also, the subheading [in here](https://mui.com/x/api/date-pickers/) should probable be **Date and Time Pickers** as well? 🤔

Ahh, okay. I thought from the issue we just wanted to update titles? If not, I’ll update all references.

**Comment by richbustos** [comment]
@LukasTy Lmk if you see other instances where "Pickers" isn't capitalized.

I updated where I feel made sense. The only files i excluded were the changelogs. Felt like it's not necessary to update those.

File: docs/pages/x/api/date-pickers/index.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Date and Time Pickers - API Reference
 
-<p class="description">This page contains an index to the most important APIs of the date pickers.</p>
+<p class="description">This page contains an index to the most important APIs of the Date Pickers.</p>
```
Comment (LukasTy) [inline docs/pages/x/api/date-pickers/index.md:3]: Probably makes sense to include the `time` pickers as they are already mentioned in the heading. 🤔 
```suggestion
<p class="description">This page contains an index to the most important APIs of the Date and Time Pickers.</p>
```

**Review by LukasTy — APPROVED** [review]
Looking great! 👍

**Pull request merged**